lib/ex_webrtc/media/ogg_reader.ex

```diff
@@ -0,0 +1,168 @@
+defmodule ExWebRTC.Media.OggReader do
+  @moduledoc """
+  Defines Ogg reader.
+
+  For now, works only with single Opus stream in the container.
+
+  Based on:
+  * [Xiph's official Ogg documentation](https://xiph.org/ogg/)
+  * [RFC 7845: Ogg Encapsulation for the Opus Audio Codec](https://www.rfc-editor.org/rfc/rfc7845.txt)
+  * [RFC 6716: Definition of the Opus Audio Codec](https://www.rfc-editor.org/rfc/rfc6716.txt)
+  """
+
+  import Bitwise
+
+  @crc_params %{
+    extend: :crc_32,
+    poly: 0x04C11DB7,
+    init: 0x0,
+    xorout: 0x0,
+    refin: false,
+    refout: false
+  }
+
+  @signature "OggS"
+  @id_signature "OpusHead"
+  @comment_signature "OpusTags"
+  @version 0
+
+  @opaque t() :: %{
+            file: File.io_device(),
+            packets: [binary()],
+            rest: binary()
+          }
+
+  @doc """
+  Opens Ogg file.
+
+  For now, works only with single Opus stream in the container.
+  This function reads the ID and Comment Headers (and, for now, ignores them).
+  """
+  @spec open(Path.t()) :: {:ok, t()} | {:error, File.posix() | :invalid_header}
+  def open(path) do
+    with {:ok, file} <- File.open(path),
+         reader <- %{file: file, packets: [], rest: <<>>},
+         # for now, we ignore ID Header and Comment Header
+         {:ok, reader, <<@id_signature, _rest::binary>>} <- do_next_packet(reader),
+         {:ok, reader, <<@comment_signature, _rest::binary>>} <- do_next_packet(reader) do
+      {:ok, reader}
+    else
+      {:error, _res} = err -> err
+      _other -> {:error, :invalid_header}
+    end
+  end
+
+  @doc """
+  Reads next Ogg packet.
+
+  One Ogg packet is equivalent to one Opus packet.
+  This function also returns the duration of the audio in milliseconds, based on Opus packet TOC sequence.
```

Comment:
Maybe let's elaborate on TOC

Reply:
I'll just redirect the reader to the RFC.